### ISSUE TITLE
ProtoUpdateListener not called for changes in nested protobuf descriptor

### ISSUE THREAD

**Issue by mauliksoneji** [gojekfarm/stencil#5]
DescriptorCache's logic to check if the protobuf descriptor has changed doesn't work for changes in nested types.
The current check to compare old and updated protobuf descriptor only compares the root level fields and doesn't check nested types.

For example:
Old descriptor:
```
message TestNestedType {
    string sample_string = 1;
    TestType test_type = 2;
}

message TestType {
    string nested_string = 1;
    int nested_int = 2;
}
```

And new descriptor:
```
message TestNestedType {
    string sample_string = 1;
    TestType test_type = 2;
}

message TestType {
    string nested_string = 1;
    int nested_int = 2;
    float nested_float = 3;
}
```

Equality check between these descriptor returns false even when the nested field has changed.

**Comment by mauliksoneji** [comment]
Currently, we are using descriptorproto's equality check to compare between old and new descriptor.
The equality check only compares the root level fields.

There can be two solutions to resolve this:
1. We implement our own comparator which compares two descriptors on nested types as well. This would make sure that `onProtoUpdate` is called if any of the nested types change.

2. We call `onProtoUpdate` when the cache is invalidated and remove the proto check. As part of this, we will leave it upto the class that implements `ProtoUpdateListener` to check if it has to take some action or not.

**Comment by mauliksoneji** [comment]
Fixed by #6
